nova_vm/src/ecmascript/builtins/data_view/abstract_operations.rs

```diff
@@ -143,3 +154,151 @@ pub(crate) fn is_view_out_of_bounds(
     // 10. Return false.
     false
 }
+
+/// ### [25.3.1.5 GetViewValue ( view, requestIndex, isLittleEndian, type )](https://tc39.es/ecma262/#sec-getviewvalue)
+///
+/// The abstract operation GetViewValue takes arguments view (an ECMAScript
+/// language value), requestIndex (an ECMAScript language value), isLittleEndian
+/// (an ECMAScript language value), and type (a TypedArray element type) and
+/// returns either a normal completion containing either a Number or a BigInt,
+/// or a throw completion. It is used by functions on DataView instances to
+/// retrieve values from the view's buffer.
+pub(crate) fn get_view_value<T: Viewable>(
+    agent: &mut Agent,
+    gc: GcScope<'_, '_>,
+    view: Value,
+    request_index: Value,
+    is_little_endian: Value,
+) -> JsResult<Value> {
+    // 1. Perform ? RequireInternalSlot(view, [[DataView]]).
+    // 2. Assert: view has a [[ViewedArrayBuffer]] internal slot.
+    let view = require_internal_slot_data_view(agent, view)?;
+
+    // 3. Let getIndex be ? ToIndex(requestIndex).
+    let get_index = to_index(agent, gc, request_index)? as usize;
```

Comment:
**suggestion:** Do an fast-path for if index and LE being integer and Boolean (or undefined) respectively. We can avoid the calls and having to pass GC token (this will become very relevant in the future.)

Reply:
Yeah, `to_boolean` can be skipped unconditionally, I forgot that it cannot trigger GC anyhow.

We can also skip the "fast path" for `to_index`. Eventually we'll want to employ those... Or maybe I'll change the APIs for `to_index` and similar kinds of functions now that I think about it... The thing is that if we know statically that we're not calling any JavaScript code, then we don't need to prepare for GC which means we don't won't need to perform any rooting of Values using `Local<Value>`. But actually indeed it's probably better to eventually turn `to_index` into something like this:

```rs
fn to_index(agent: &mut Agent, index: Value, on_needs_gc: FnMut(&mut Agent, Object) -> JsResult<Number>) -> JsResult<Number> { ... }
```

... well, not this exactly. But something where effectively the caller provides a callback that the method can use to signal that it is about to call into JavaScript and the caller should store its Values into Locals.